packages/graph/hash_graph/lib/graph/src/store/postgres/query/property_type.rs

```diff
@@ -28,104 +29,37 @@ impl PostgresQueryRecord for PropertyType {
 impl Path for PropertyTypeQueryPath {
     fn relations(&self) -> Vec<Relation> {
         match self {
-            Self::BaseUri | Self::Version => vec![Relation {
-                current_column_access: Self::VersionId.column_access(),
-                join_table_name: TableName::TypeIds,
-                join_column_access: Self::VersionId.column_access(),
-            }],
-            Self::DataTypes(path) => [
-                Relation {
-                    current_column_access: Self::VersionId.column_access(),
-                    join_table_name: TableName::PropertyTypeDataTypeReferences,
-                    join_column_access: ColumnAccess::Table {
-                        column: "source_property_type_version_id",
-                    },
-                },
-                Relation {
-                    current_column_access: ColumnAccess::Table {
-                        column: "target_data_type_version_id",
-                    },
-                    join_table_name: TableName::DataTypes,
-                    join_column_access: DataTypeQueryPath::VersionId.column_access(),
-                },
-            ]
-            .into_iter()
-            .chain(path.relations())
-            .collect(),
-            Self::PropertyTypes(path) => [
-                Relation {
-                    current_column_access: Self::VersionId.column_access(),
-                    join_table_name: TableName::PropertyTypePropertyTypeReferences,
-                    join_column_access: ColumnAccess::Table {
-                        column: "source_property_type_version_id",
-                    },
-                },
-                Relation {
-                    current_column_access: ColumnAccess::Table {
-                        column: "target_property_type_version_id",
-                    },
-                    join_table_name: TableName::PropertyTypes,
-                    join_column_access: Self::VersionId.column_access(),
-                },
-            ]
-            .into_iter()
-            .chain(path.relations())
-            .collect(),
+            Self::BaseUri | Self::Version => vec![Relation::PropertyTypeIds],
```

Comment:
The old version of the code seems to do a join. Is this handled by the `Relation` now?

Reply:
Yes, `Relation` holds the information which tables needs to be joined